test/run-versions

```diff
@@ -4,59 +4,58 @@
 # shellcheck source=bin/default_pythons
 source "bin/default_pythons"
 
-testPythonDefault() {
-  updateVersion "pythonDefault" $DEFAULT_PYTHON_VERSION
-  compile "pythonDefault"
-  assertCaptured $DEFAULT_PYTHON_VERSION
+testPythonVersionUnspecified() {
+  compile "python_version_unspecified"
+  assertCaptured "Installing ${DEFAULT_PYTHON_VERSION}"
   assertNotCaptured "security update"
   assertCaptured "Installing pip 20.1.1, setuptools 47.1.1 and wheel 0.34.2"
   assertCaptured "Installing SQLite3"
   assertCapturedSuccess
 }
 
-testPython2() {
-    updateVersion "python2" $LATEST_27
-    echo $LATEST_27 > "runtime.txt"
-    compile "python2"
-    assertCaptured $LATEST_27
-    assertCaptured "python-2-7-eol-faq";
-    assertNotCaptured "security update"
-    assertCaptured "Installing pip 20.1.1, setuptools 44.1.1 and wheel 0.34.2"
-    assertCaptured "Installing SQLite3"
-    assertCapturedSuccess
+testPython2_7() {
+  compile "python2"
+  assertCaptured "Installing ${LATEST_27}"
+  assertCaptured "python-2-7-eol-faq";
+  assertNotCaptured "security update"
+  assertCaptured "Installing pip 20.1.1, setuptools 44.1.1 and wheel 0.34.2"
+  assertCaptured "Installing SQLite3"
+  assertCapturedSuccess
 }
 
-testPython2_warn() {
-    compile "python2_warn"
-    assertCaptured "python-2.7.15"
-    assertCaptured "python-2-7-eol-faq";
-    assertCaptured "Only the latest version"
-    assertCaptured "Installing SQLite3"
-    assertCapturedSuccess
+testPython2_7_warn() {
+  compile "python2_warn"
+  assertCaptured "Installing python-2.7.15"
+  assertCaptured "python-2-7-eol-faq";
+  assertCaptured "Only the latest version"
+  assertCaptured "${LATEST_27}"
+  assertCapturedSuccess
 }
 
 testPython3_4() {
   compile "python3_4"
-  assertCaptured $LATEST_34
+  assertCaptured "Installing ${LATEST_34}"
   assertNotCaptured "security update"
   assertCaptured "Installing pip 19.1.1, setuptools 43.0.0 and wheel 0.33.6"
+  assertCaptured "Installing SQLite3"
   # Can't use `assertCapturedSuccess` since Pip outputs a Python 3.4 EOL warning to stderr,
   # and the newest Pip that works on Python 3.4 doesn't support `PIP_NO_PYTHON_VERSION_WARNING`.
   assertCapturedSuccessWithStdErr
 }
 
 testPython3_4_warn() {
   compile "python3_4_warn"
-  assertCaptured "python-3.4.9"
+  assertCaptured "Installing python-3.4.9"
   assertCaptured "security update!"
+  assertCaptured "${LATEST_34}"
```

Comment:
Can I double-check with you about this and the corresponding test for 3.5? It doesn't make sense to me to include both a check for the `security update!` string and the most recent minor version. Running `make test` locally complains at this line and the test for 3.5 at line 58.

Reply:
@vanastassiou Hi! In the case of a security update being available, the warning message shown says:

> Python has released a security update! Please consider upgrading to `<latest version>`

This is what that assertion is checking against.

See:
https://github.com/heroku/heroku-buildpack-python/blob/fcf696b835d7abce1631d841de7ae5fdd5dde45c/bin/steps/python#L46-L59

The tests for this repo are passing on `main`:
https://travis-ci.com/github/heroku/heroku-buildpack-python

Are you using an older branch/fork, or else have set a custom `BUILDPACK_S3_BASE_URL` pointing at a bucket that's missing those files?